src/test/java/com/intuit/fuzzymatcher/component/MatchServicePerfTest.java

```diff
@@ -81,4 +91,39 @@ static String getAddress(String[] csv) {
         }
         return address;
     }
+
+    private static CSVReader getCSVReader(String FileName) throws FileNotFoundException {
+        return new CSVReaderBuilder(
+                new FileReader(MatchServicePerfTest.class.getClassLoader().getResource(FileName).getFile()))
+                .withSkipLines(1)
+                .build();
+    }
+
+    private List<Document> readTxnFromCSV(String fileName, AtomicInteger index) throws FileNotFoundException {
+
+        List<Document> list = new ArrayList<>();
+        StreamSupport.stream(getCSVReader(fileName).spliterator(), false).forEach(csv -> {
+            int i =0;
+            list.add(new Document.Builder(index.getAndIncrement()+"")
+                    .addElement(new Element.Builder().setValue(csv[0]).setType(TEXT).setVariance(i++ + "").createElement())
+                    .addElement(new Element.Builder().setValue(csv[1]).setType(TEXT).setVariance(i++ + "").createElement())
+                    .addElement(new Element.Builder().setValue(csv[2]).setType(TEXT).setVariance(i++ + "").createElement())// should be date
+                    .addElement(new Element.Builder().setValue(csv[3]).setType(NUMBER).setVariance(i++ + "").createElement())
+                    .addElement(new Element.Builder().setValue(csv[4]).setType(NUMBER).setVariance(i++ + "").createElement())
+                    .addElement(new Element.Builder().setValue(csv[5]).setType(NUMBER).setVariance(i++ + "").createElement())
+//                    .addElement(new Element.Builder().setValue(csv[6]).setType(NUMBER).setVariance(i++ + "").createElement())
+//                    .addElement(new Element.Builder().setValue(csv[7]).setType(NUMBER).setVariance(i++ + "").createElement())
+//                    .addElement(new Element.Builder().setValue(csv[8]).setType(NUMBER).setVariance(i++ + "").createElement())
+                    .createDocument());
+        });
+        return list;
+    }
+
+    @Test
+    public void matchBigTransactions() throws FileNotFoundException {
+        AtomicInteger atomicInteger =  new AtomicInteger();
+        List<Document> govt = readTxnFromCSV("Txns.csv", atomicInteger);
+        List<Document> book = readTxnFromCSV("Txns.csv", atomicInteger);
+        applyMatch(book, govt);
```

Comment:
I see you are matching the same document against itself.  You could use the ```matchService.applyMatch(List<Document> documents)``` method where you only need to pass a single instance of the document , vs ```matchService.applyMatch(List<Document> documents, List<Document> matchWith)``` 
I am assuming you are doing this just to generate load ?